install_sct

```diff
@@ -403,22 +411,29 @@ print info "
 fetch_os_type
 check_requirements
 
-# Catch SCT version
+# Check to see if the PWD contains the project source files (using `version.txt` as a proxy for the entire source dir)
+# If it exists, then we can reliably access source files (`version.txt`, `requirements-freeze.txt) from the PWD.
 if [[ -e "spinalcordtoolbox/version.txt" ]]; then
   SCT_VERSION="$(cat spinalcordtoolbox/version.txt)"
-else
-  die "ERROR: version.txt not found."
-fi
-
-# Get installation type (from git or from package) if not already specified
-if [[ -z "$SCT_INSTALL_TYPE" ]]; then
-  # The file 'requirements-freeze.txt` only exists on the `release` branch
-  # or in stable release "Source Code" .zips (which point at tags `release`)
-  if [[ -e "requirements-freeze.txt" ]]; then
-    SCT_INSTALL_TYPE="package"   # i.e. release installations
-  else
-    SCT_INSTALL_TYPE="in-place"  # i.e. dev installations
+  # Get installation type if not already specified
+  if [[ -z "$SCT_INSTALL_TYPE" ]]; then
+    # The file 'requirements-freeze.txt` only exists for stable releases
+    if [[ -e "requirements-freeze.txt" ]]; then
+      SCT_INSTALL_TYPE="package"
+    # If it doesn't exist, then we can assume that a dev is performing an in-place installation from master
+    else
+      SCT_INSTALL_TYPE="in-place"
+    fi
   fi
+# If version.txt isn't present, then the installation script is being run by itself (i.e. without source files).
+# So, we need to clone SCT to a TMPDIR to access the source files, and update SCT_SOURCE accordingly.
+else
+  SCT_SOURCE="$TMP_DIR/spinalcordtoolbox"
+  echo "Source files not present. Cloning source files to $SCT_SOURCE."
+  git clone -b "$SCT_GIT_REF" --single-branch --depth 1 https://github.com/spinalcordtoolbox/spinalcordtoolbox.git "$SCT_SOURCE"
+  SCT_VERSION=$(cat "$SCT_SOURCE/spinalcordtoolbox/version.txt")
+  # Since we're git cloning into a TMPDIR, this can never be an "in-place" installation, so we force "package" instead.
+  SCT_INSTALL_TYPE="package"
```

Comment:
Would it be simpler to `git clone` into the current directory? A quick test on my machine shows that `git` will happily do so.
```suggestion
# So, we need to clone SCT to access the source files.
else
  echo "Source files not present. Cloning source files."
  git clone -b "$SCT_GIT_REF" --single-branch --depth 1 https://github.com/spinalcordtoolbox/spinalcordtoolbox.git .
  SCT_VERSION=$(cat "spinalcordtoolbox/version.txt")
```
(The above suggestion isn't a complete implementation of this, though.)

Reply:
If you do this, I would suggest keeping the `version.txt` test for the `git clone` logic, and then doing the rest of the logic (setting `SCT_VERSION`, checking for `requirements-freeze.txt`, etc.) unconditionally, afterwards.

Reply:
There are a couple of factors here that make this a bit tricky, I think?

- A user may download the install script, and then run it by double-clicking on it (in much the same way they're instructed to do for the Windows batch scripts). In that case, the current working directory may be `Downloads/`, `Desktop/`, etc. 
- Additionally, the script distinguishes between `$SCT_SOURCE` and `SCT_DIR`:
    - `$SCT_SOURCE`, in this case, would be the PWD, because it contains the source files.
    - `$SCT_DIR` is the directory that the user has chosen to install SCT to (typically e.g. `~/sct_5.8/`)
    - Once `$SCT_DIR` has been chosen, the source files are copied from `$SCT_SOURCE` to `$SCT_DIR`.
- So, if we were to clone into `.`, would the copy step inadvertently copy user files from, say, `Downloads/` to `$SCT_DIR`?

The one other benefit to using a tmpdir (as opposed to, say, `./spinalcordtoolbox`) is that it avoids the risk cloning to a non-empty folder, since the generated tmpdir will be unique on each repeated run of `./install_sct`.